### PR TITLE
Alligned spring-session and spring versions.

### PR DESCRIPTION
**Pull request by irinaepshteyn**

### CODE REVIEW

**Comment by irinaepshteyn** [comment]
@metacubed , to make your review easier, UaaDefaultCookieSerializer.java I added is a combination between [DefaultCookieSerializer.java from spring-session 2.3.1.RELEASE](https://github.com/spring-projects/spring-session/blame/2.3.1.RELEASE/spring-session-core/src/main/java/org/springframework/session/web/http/DefaultCookieSerializer.java) and [DefaultCookieSerializer.java from spring-session 1.3.5.RELEASE](https://github.com/spring-projects/spring-session/blob/1.3.5.RELEASE/spring-session/src/main/java/org/springframework/session/web/http/DefaultCookieSerializer.java) to add SameSite support.

Also, degraded tests are failing in Jenkins and passing locally.  I suspect it's environmental; takes longer than expected to redirect.

File: server/src/test/java/org/cloudfoundry/identity/uaa/web/beans/UaaDefaultCookieSerializerTest.java [hunk 1]
```diff
@@ -0,0 +1,493 @@
+package org.cloudfoundry.identity.uaa.web.beans;
```
Comment (metacubed) [inline server/src/test/java/org/cloudfoundry/identity/uaa/web/beans/UaaDefaultCookieSerializerTest.java:1]: Do you need license headers here?

File: server/src/main/java/org/cloudfoundry/identity/uaa/web/beans/UaaDefaultCookieSerializer.java [hunk 2]
```diff
@@ -0,0 +1,434 @@
+package org.cloudfoundry.identity.uaa.web.beans;
```
Comment (metacubed) [inline server/src/main/java/org/cloudfoundry/identity/uaa/web/beans/UaaDefaultCookieSerializer.java:1]: Do you need license headers here?

File: server/src/main/java/org/cloudfoundry/identity/uaa/web/beans/UaaDefaultCookieSerializer.java [hunk 3]
```diff
@@ -0,0 +1,434 @@
...
+        try {
+            ServletRequest.class.getMethod("startAsync");
+            return true;
+        }
+        catch (NoSuchMethodException e) {
+        }
+        return false;
+    }
+
+}
```
Comment (metacubed) [inline server/src/main/java/org/cloudfoundry/identity/uaa/web/beans/UaaDefaultCookieSerializer.java:434]: Minor issue, add a newline at the end of the file.

**Review by metacubed — APPROVED** [review]

File: server/src/main/java/org/cloudfoundry/identity/uaa/web/beans/UaaDefaultCookieSerializer.java (hunk 2 above)
Comment (irinaepshteyn) [reply server/src/main/java/org/cloudfoundry/identity/uaa/web/beans/UaaDefaultCookieSerializer.java:1]: UAA and spring use the same apache license: "Licensed under the Apache License, Version 2.0 (the "License");"
We don't provide any additional license headers for predix customizations; keep Pivotal headers wherever present.
Based on that decided not to add a header

File: server/src/main/java/org/cloudfoundry/identity/uaa/web/beans/UaaDefaultCookieSerializer.java (hunk 3 above)
Comment (irinaepshteyn) [reply server/src/main/java/org/cloudfoundry/identity/uaa/web/beans/UaaDefaultCookieSerializer.java]: Done

**Pull request merged**